component/main.jsonnet

```diff
@@ -129,9 +129,62 @@ local organizationNamespaces = kyverno.ClusterPolicy('organization-namespaces')
   },
 };
 
+/**
+  * ClusterRole allowed to create and edit reserved namespaces
+  */
+local editReservedNamespacesClusterRole = kube.ClusterRole(params.reservedNamespaces.allowedClusterRoles.name) {
+  rules: [],
+};
+
+/**
+  * Refute create or edit reserved namespaces
+  * This policy will:
+  * - Check if namespace name matches one of the disallowed namespace patterns
+  * - Check if user has cluster role that allows them to create reserved namespaces
+  * - Deny namespace creation or modification
+  */
+local refuteReservedNamespaces = kyverno.ClusterPolicy('refute-reserved-namespaces') {
+  spec: {
```

Comment:
In #9 I have added a set of [recommended labels](https://syn.tools/syn/references/labels-guide.html). Maybe we can consolidate this? I tried it with a lib/libsonnet file but somehow couldn't make use of it from the same component :shrug:

Reply:
I added a `component/common.libsonnet` with the labels.